changelog/b6227.dd

```diff
@@ -0,0 +1,24 @@
+Comparison of values belonging to different enums is deprecated.
+
+Since the comparison is done by comparing the value of the enumeration members
+we give up on the type-safety of the enumeration type possibly hiding a whole
+class of bugs that could be caught at compile time.
+
+This change doesn't affect the anonymous enumerations.
+
+---
+enum Status
+{
+    GOOD,
+    BAD
+}
+enum OtherStatus
+{
+    OK
+    NO,
```

Comment:
We should follow the usual convention here, i.e. enum values in lowercase. Great example btw.